quinn-proto/src/connection/streams/send.rs

```diff
@@ -51,7 +50,7 @@ impl Send {
         }
     }
 
-    pub(super) fn write(&mut self, data: &[u8]) -> Result<usize, WriteError> {
+    pub(super) fn write<S: BytesSource>(&mut self, source: &mut S) -> Result<usize, WriteError> {
```

Comment:
Should just this return `()` for `Ok`, since the caller can get the consumption from the `BytesSource` itself?

Reply:
Yeah. I just kept it because `Streams` directly consumes the return value. But probably it's better to switch for `()` for avoiding duplicate information.